config/routes.rb

```diff
@@ -185,15 +191,15 @@
   # If parameters are used that are the same as those in the search form, pagination with kaminari will break
   match "teaching-jobs-in-:location_facet",
         to: "vacancies#index", as: :location, via: :get,
-        constraints: ->(request) { LocationPolygon.include?(request.params[:location_facet]) }
+        constraints: ->(request) { LocationPolygon.include?(request.params[:location_facet].tr("-", " ")) }
 
   match "teaching-jobs-for-:job_role",
         to: "vacancies#index", as: :job_role, via: :get,
-        constraints: ->(request) { Vacancy.job_roles.key?(request.params[:job_role]) },
+        constraints: ->(request) { Vacancy.job_roles.key?(request.params[:job_role].tr("-", "_")) },
```

Comment:
In the review app when I click on:

<img width="532" alt="Screenshot 2021-09-28 at 13 27 13" src="https://user-images.githubusercontent.com/30624173/135086425-cb91607e-35ef-439e-9c39-58308e209eb4.png">
 
The URL is still teaching-jobs-for-ect_suitable.